exporter/clickhouseexporter/config_test.go

```diff
@@ -243,7 +244,76 @@ func TestConfig_buildDSN(t *testing.T) {
 			args: args{
 				database: defaultDatabase,
 			},
-			want: "tcp://127.0.0.1:9000/default",
+			want: "tcp://127.0.0.1:9000/default?async_insert=true",
+		},
+		{
+			name: "when config option is missing, preserve async_insert false in DSN",
+			fields: fields{
+				Endpoint: "tcp://127.0.0.1:9000?async_insert=false",
+			},
+			want: "tcp://127.0.0.1:9000/default?async_insert=false",
+		},
+		{
+			name: "when config option is missing, preserve async_insert true in DSN",
+			fields: fields{
+				Endpoint: "tcp://127.0.0.1:9000?async_insert=true",
+			},
+			want: "tcp://127.0.0.1:9000/default?async_insert=true",
+		},
+		{
+			name: "ignore config option when async_insert is present in connection params as false",
+			fields: fields{
+				Endpoint:         "tcp://127.0.0.1:9000?async_insert=false",
+				ConnectionParams: map[string]string{"async_insert": "false"},
+				AsyncInsert:      configBool(true),
+			},
+
+			want: "tcp://127.0.0.1:9000/default?async_insert=false",
+		},
+		{
+			name: "ignore config option when async_insert is present in connection params as true",
+			fields: fields{
+				Endpoint:         "tcp://127.0.0.1:9000?async_insert=false",
+				ConnectionParams: map[string]string{"async_insert": "true"},
+				AsyncInsert:      configBool(false),
+			},
+
+			want: "tcp://127.0.0.1:9000/default?async_insert=true",
+		},
+		{
+			name: "ignore config option when async_insert is present in DSN as false",
+			fields: fields{
+				Endpoint:    "tcp://127.0.0.1:9000?async_insert=false",
+				AsyncInsert: configBool(true),
+			},
+
+			want: "tcp://127.0.0.1:9000/default?async_insert=false",
+		},
+		{
+			name: "use async_insert true config option when it is not present in DSN",
+			fields: fields{
+				Endpoint:    "tcp://127.0.0.1:9000",
+				AsyncInsert: configBool(true),
+			},
+
+			want: "tcp://127.0.0.1:9000/default?async_insert=true",
+		},
+		{
+			name: "use async_insert false config option when it is not present in DSN",
+			fields: fields{
+				Endpoint:    "tcp://127.0.0.1:9000",
+				AsyncInsert: configBool(false),
+			},
+
+			want: "tcp://127.0.0.1:9000/default?async_insert=false",
+		},
+		{
+			name: "set async_insert to true when not present in config or DSN",
```

Comment:
It would have to be done in its own test, but it would be good to add another test that asserts that the final endpoint is the same whether `async_insert` is coming through from the `endpoint` config option, from the `connection_params`, or from a combination of `async_insert` and `endpoint`. This way we can ensure going forward that the final result is the same from each configuration path.